applications/crossbar/src/modules/cb_local_provisioner_templates.erl

```diff
@@ -144,81 +147,104 @@ resource_exists(_, ?IMAGE_REQ) ->
     load_provisioner_template_summary(Context);
-validate(#cb_context{req_verb = ?HTTP_PUT}=Context) ->
+validate_verb(Context, ?HTTP_PUT) ->
     create_provisioner_template(Context).
 
-validate(#cb_context{req_verb = ?HTTP_GET}=Context, DocId) ->
+validate_verb(Context, ?HTTP_GET, DocId) ->
     load_provisioner_template(DocId, Context);
-validate(#cb_context{req_verb = ?HTTP_POST}=Context, DocId) ->
+validate_verb(Context, ?HTTP_POST, DocId) ->
     update_provisioner_template(DocId, Context);
-validate(#cb_context{req_verb = ?HTTP_DELETE}=Context, DocId) ->
+validate_verb(Context, ?HTTP_DELETE, DocId) ->
     load_provisioner_template(DocId, Context).
 
-validate(#cb_context{req_verb = ?HTTP_GET}=Context, DocId, ?IMAGE_REQ) ->
+validate_verb(Context, ?HTTP_GET, DocId, ?IMAGE_REQ) ->
     load_template_image(DocId, Context);
-validate(#cb_context{req_verb = ?HTTP_PUT}=Context, DocId, ?IMAGE_REQ) ->
-    upload_template_image(DocId, Context);
-validate(#cb_context{req_verb = ?HTTP_POST}=Context, DocId, ?IMAGE_REQ) ->
-    upload_template_image(DocId, Context);
-validate(#cb_context{req_verb = ?HTTP_DELETE}=Context, DocId, ?IMAGE_REQ) ->
+validate_verb(Context, ?HTTP_PUT, _DocId, ?IMAGE_REQ) ->
+    upload_template_image(Context);
+validate_verb(Context, ?HTTP_POST, _DocId, ?IMAGE_REQ) ->
+    upload_template_image(Context);
+validate_verb(Context, ?HTTP_DELETE, DocId, ?IMAGE_REQ) ->
     load_template_image(DocId, Context).
 
--spec post(#cb_context{}, path_token()) -> #cb_context{}.
-post(#cb_context{doc=JObj}=Context, DocId) ->
+-spec post(cb_context:context(), path_token()) -> cb_context:context().
+post(Context, DocId) ->
     %% see note at top of file
+    JObj = cb_context:doc(Context),
     Template = wh_json:get_value(<<"template">>, JObj),
     Doc = wh_json:delete_key(<<"template">>, JObj),
-    case crossbar_doc:save(Context#cb_context{doc=Doc}) of
-        #cb_context{resp_status=success, resp_data=SavedResp}=Context1 ->
-            Opts = [{headers, [{content_type, "application/json"}]}],
-            case crossbar_doc:save_attachment(DocId, ?TEMPLATE_ATTCH, wh_json:encode(Template), Context, Opts) of
-                #cb_context{resp_data=success} ->
-                    Context1#cb_context{resp_data=wh_json:set_value(<<"template">>, Template, SavedResp)};
+    Context1 = crossbar_doc:save(cb_context:set_doc(Context, Doc)),
+    case cb_context:resp_status(Context1) of
+        'success' ->
+            SavedResp = cb_context:resp_data(Context1),
+            Opts = [{'headers', [{'content_type', "application/json"}]}],
+            Ctx2 = crossbar_doc:save_attachment(DocId, ?TEMPLATE_ATTCH, wh_json:encode(Template), Context, Opts),
+            case cb_context:resp_status(Ctx2) of
+                'success' ->
+                    cb_context:set_resp_data(Context1, wh_json:set_value(<<"template">>, Template, SavedResp));
                 Else -> Else
             end;
         Else -> Else
```

Comment:
This should now return Context1 instead of Else, same with line 200 above (Ctx2 that is)